internal/pkg/runtime/launcher/oci/launcher_linux.go

```diff
@@ -219,7 +219,15 @@ func (l *Launcher) createSpec() (*specs.Spec, error) {
 	if err != nil {
 		return nil, err
 	}
-	spec.Mounts = mounts
+	spec.Mounts = append(spec.Mounts, mounts...)
+
+	for _, cdiDevice := range l.cfg.CDIDevices {
```

Comment:
Although this is a small and simple loop, please could you pull it out into a function so it ends up as `getCDIMounts` returning a slice to append. Ideally in a `cdi_linux.go` file. It's just a preference for separating stuff a bit more as this launcher code is going to see a lot of work, and expansion for 4.0.

Reply:
Actually - I see why the above isn't easy, sorry.... there's only a path to mutate the spec.

Let's go for an `addCDIDevices` function, then... which will take the spec and modify it.

Reply:
Yeah - per my follow-up, ignore that. I had thought it might be easier to get mounts, env vars etc out individually.. but I should have checked in the CDI code first.